fiftyone/core/annotation/attributes.py

```
@@ -0,0 +1,216 @@
+        d[name] = value
+    return d
+
+
+def _require_keys(d: dict, keys: tuple, cls: type) -> None:
+    """Raises ``ValueError`` if any of ``keys`` are missing from ``d``."""
+    missing = [k for k in keys if k not in d]
+    if missing:
+        raise ValueError(
+            f"{cls.__name__} dict missing required key(s): "
+            f"{', '.join(missing)}"
+        )
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)